### ISSUE TITLE
Change the way weapon restrictions are implemented to destroy weapons

### ISSUE THREAD

**Issue by RemyLebeau** [War3Source/War3Source#154]
**TL;DR**: Change it so that when you put in weapon restrictions instead of forcing the player to drop the weapon it actually destroys it

**Background**: We run a mini war3 game on our server called duckhunt, which involved Ts going Druid of the Talon and CTs going a shotgun only race.  Druid of the talon swaps between normal and a bird figure that makes you drop your knife.

**Problem**: When an entire team is constantly swapping back and forth, being given a new knife then dropping it when transforming, it creates a lot of entities that are being dropped around the map, causing some lag issues.

**Solution**: Change it so that instead of dropping the weapons, we destroy them.

Replace 
<code>
CS_DropWeapon(client,wpnent,true);</code>

For something like (from SMLib):
<code>
        if (RemovePlayerItem(client, weapon)) {
            AcceptEntityInput(weapon, "kill");
        }</code>

**Difficulty**: This would mean that when anyone with weapon restrictions runs over a weapon that is on the ground, they will destroy it...  This is bad, since it means non-weapon restricted races are deprived of a source of weapons.  I'm not completely sure how to address this issue...

**Comment by necavi** [comment]
Which races seem to be causing this issue?
